actors/actor.go

```diff
@@ -0,0 +1,48 @@
+/*interface file for actors*/
+package actors
+
+import (
+	"fmt"
+	"student20_pop/db"
+	"student20_pop/define"
+)
+
+type Actor interface {
+	//Public functions
+	HandleWholeMessage(msg []byte, userId int) ([]byte, []byte, []byte)
+	//Private functions
+	handlePublish(generic define.Generic) ([]byte, []byte, error)
+	handleCreateLAO(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleUpdateProperties(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleWitnessMessage(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleLAOState(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+	handleCreateRollCall(message define.Message, channel string, generic define.Generic) ([]byte, []byte, error)
+}
+
+//general actors functions, act only in the "Sub" database
+func handleSubscribe(generic define.Generic, userId int) error {
+	params, err := define.AnalyseParamsLight(generic.Params)
```

Comment:
Is this clearer? params, err := parser.ParseParams(query.Params)
It still has the issue of parsing params to return params, but the idea is that the field query.Params is in json, while the returned params is a structure

Reply:
it might make sense to rename it like you suggest.

In a general sense, compare the following things.
```
params, err := define.AnalyseParamsLight(generic.Params)
``` 
to that:
```
params, err := jsonrpc.ParseParams(rpc.Params)
```
I think the latter fully expresses what you're doing, 

Obviously, I write the latter freely, unencumbered by the constraints of your codebase, just to express what I mean.
In a general sense, however, you should aim to achieve that kind of readability.
Code is written a couple of times and read a few dozens or hundreds of time. You're doing your future self a favor :)